packages/atclient/src/string_utils.c

```diff
@@ -59,6 +59,25 @@ bool atclient_string_utils_ends_with(const char *string, const char *suffix) {
 
 }
 
+int atclient_string_utils_get_substring_position(const char* string, const char* substring, char** position) {
+  int ret = -1;
+  if(strlen(substring) > strlen(string)) {
+    ret = -1;
+    goto exit;
+  }
+  if(position == NULL) {
+    ret = -1;
+    goto exit;
+  }
+  *position =  strstr(string, substring);
+  if(*position == NULL) {
```

Comment:
Can you verify this call please?

position should be a pointer to an integer meaning that `*position == NULL` will be true if the integer value is zero (NULL = 0 on most (all?) platforms)

I think you mean `position == NULL` here

Reply:
We do not deal with any integers here. The method _strstr()_ returns a pointer to the location of the substring. So in this case, _position_ is a pointer to the character 'd'(as in _data:_) wherever it is located in the given string. So *position would actually resolve to the string starting with _data:_ until terminated. If you find the variable name _position_ misleading LMK, so that I can find an alternative that is better suited.

Reply:
position is a double pointer, so *position is actually a pointer (points to 'd' in data:). **position would resolve to the letter 'd'. This would only return true if *position does not point to anything.